habitat/sims/pyrobot/pyrobot.py

```diff
@@ -0,0 +1,281 @@
+                sensor_cfg.TYPE
+            )
+            robot_sensors.append(sensor_type(sensor_cfg))
+        self._sensor_suite = SensorSuite(robot_sensors)
+
+        config_pyrobot = {
+            "base_controller": self._config.BASE_CONTROLLER,
+            "base_planner": self._config.BASE_PLANNER,
+        }
+
+        assert (
+            self._config.ROBOT in self._config.ROBOTS
+        ), "Invalid robot type {}".format(self._config.ROBOT)
+        self._robot_config = getattr(self._config, self._config.ROBOT.upper())
+
+        action_spaces_dict = {}
+
+        self._action_space = self._robot_action_space(
+            self._config.ROBOT, self._robot_config
+        )
+
+        self._robot = pyrobot.Robot(
+            self._config.ROBOT, base_config=config_pyrobot
+        )
+
+    def get_robot_observations(self):
+        return {
+            "rgb": self._robot.camera.get_rgb(),
+            "depth": self._robot.camera.get_depth(),
+            "bump": self._robot.base.base_state.bumper,
+        }
+
+    @property
+    def sensor_suite(self) -> SensorSuite:
+        return self._sensor_suite
+
+    @property
+    def base(self):
+        return self._robot.base
+
+    @property
+    def camera(self):
+        return self._robot.camera
+
+    def _robot_action_space(self, robot_type, robot_config):
+        action_spaces_dict = {}
+        for action in robot_config.ACTIONS:
+            action_spaces_dict[action] = ACTION_SPACES[robot_type.upper()][
+                action
+            ]
+        return spaces.Dict(action_spaces_dict)
+
+    @property
+    def action_space(self) -> Space:
+        return self._action_space
+
+    def reset(self):
+        self._robot.camera.reset()
+
+        robot_obs = self.get_robot_observations()
+        observations = self._sensor_suite.get_observations(robot_obs)
+        return observations
+
+    def step(self, action, action_params):
+        r"""Step in reality. Currently the supported
+        actions are the ones defined in `_locobot_base_action_space`
+        and `_locobot_camera_action_space`. For details on how
+        to use these actions please refer to the documentation
+        of namesake methods in PyRobot
+        (https://github.com/facebookresearch/pyrobot).
+        """
+        if action in self._robot_config.BASE_ACTIONS:
+            getattr(self._robot.base, action)(**action_params)
+        elif action in self._robot_config.CAMERA_ACTIONS:
+            getattr(self._robot.camera, action)(**action_params)
+        else:
+            raise ValueError("Invalid action {}".format(action))
+
+        observations = self._sensor_suite.get_observations(
+            self.get_robot_observations()
+        )
+
+        return observations
+
+    def render(self, mode: str = "rgb") -> Any:
+        robot_obs = self.get_robot_observations()
+        observations = self._sensor_suite.get_observations(robot_obs)
```

Comment:
If we add one more PyRobot sensor type that doesn't need `robot_obs` argument it will start failing. That's important to use keyword arguments using Suites:  
```suggestion
        observations = self._sensor_suite.get_observations(robot_obs=robot_obs)
```